tower-buffer/src/lib.rs

```diff
@@ -337,21 +379,21 @@ where
 
             match self.state {
                 Full => {
-                    return Err(Error::Full);
+                    return Err(FullError.into());
                 }
                 Failed(ref e) => {
-                    return Err(Error::Closed(e.clone()));
+                    return Err(ClosedError(e.clone()).into());
```

Comment:
TIOLI: you could avoid having to explicitly call `into` here (and elsewhere) if you used the `?` operator instead:
```rust
Failed(ref e) => Err(ClosedError(e.clone()))?,
```

Reply:
Oh, that'd be a good improvement. Most of the stuff I was doing here are _mostly_ mechanical.